pkg/base1/test-echo.js

```diff
@@ -79,41 +79,55 @@ QUnit.test("binary", function (assert) {
     channel.send(buffer);
 });
 
-// This is implemented in the C bridge, but not in Python.
-// Nobody uses it, so skip the test for now.
-QUnit.test.skip("fence", function (assert) {
+const test_server = {
+    address: window.location.hostname,
+    port: parseInt(window.location.port)
+};
+
+QUnit.test("fence", function (assert) {
     const done = assert.async();
-    assert.expect(2);
 
-    const before = cockpit.channel({ payload: "echo" });
-    before.addEventListener("message", onMessage);
-
-    const fence = cockpit.channel({ payload: "echo", group: "fence" });
-    fence.addEventListener("message", onMessage);
-
-    const after = cockpit.channel({ payload: "echo" });
-    after.addEventListener("message", onMessage);
-
-    const received = [];
-    function onMessage(ev, payload) {
-        received.push(payload);
-        if (received.length == 3) {
-            assert.deepEqual(received, ["1", "2", "3"], "got back before and fence data");
-            fence.close();
-        } else if (received.length == 5) {
-            assert.deepEqual(received, ["1", "2", "3", "4", "5"], "got back data in right order");
-            before.close();
-            after.close();
-            done();
-        }
-    }
-
-    /* We send messages in this order, but they should echoed in numeric order */
-    before.send("1");
-    after.send("4");
-    after.send("5");
-    fence.send("2");
-    fence.send("3");
+    // This is implemented in the C bridge, but not in Python.
+    cockpit.http(test_server).get("/mock/info")
```

Comment:
Less chaining and arrows.  More `await`/`async`, please.

Reply:
Done.